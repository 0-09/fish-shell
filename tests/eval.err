
####################
# Testing eval builtin
fish: Unexpected end of string, expecting ')'
(
^

fish: The expanded command was empty.
""
^

####################
# Testing eval with empty functions, blocks, and arguments
