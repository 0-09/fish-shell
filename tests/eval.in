logmsg "Testing eval builtin"

# Regression test for issue #4443
eval set -l previously_undefined foo
echo $previously_undefined

# Test redirection
eval "echo you can\\'t see this 1>&2" 2>/dev/null

# Test return statuses
false; eval true; echo $status # 0
false; eval false; echo $status # 1

# Test return status in case of parsing error
false; eval "("; echo $status # 1
false; eval '""'; echo $status # 1

function empty
end

# Regression tests for issue #5692
logmsg "Testing eval with empty functions, blocks, and arguments"
false; eval;
echo blank eval: $status # 0

false; eval "";
echo empty arg eval: $status # 0

false; eval empty;
echo empty function eval $status # 0

false; eval "begin; end;";
echo empty block eval: $status # 0
