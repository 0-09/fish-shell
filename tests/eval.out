
####################
# Testing eval builtin
foo
0
1
1
1

####################
# Testing eval with empty functions, blocks, and arguments
blank eval: 0
empty arg eval: 0
empty function eval 0
empty block eval: 0
